File: day_2/factorial.js
//Solution One
function factorial(number) {
  if (number<0) {
    return "Please provide a positive integer."
  } else if (number ===1 | number ===0) {
    return 1
  } else if (number > 1){
  return (BigInt(number) * BigInt(factorial(number-1))).toString().replace("n", "")
  }
}

console.log(factorial(8))
console.log(factorial(18))
console.log(factorial(45))

//Solution Two
function second_factorial(number) {
  if (number<0) {
    return "Please provide a positive integer."
  }
  let result = BigInt(1)
  for (let i=1; i<=number; i++) {
    result *= BigInt(i)
  }
  return result.toString().replace("n", "")
}

console.log(second_factorial(8))
console.log(second_factorial(18))
console.log(second_factorial(45))

File: day_2/factorial.py
#Solution One
import math

def factorial_of_number(number):
  return math.factorial(number)

print(factorial_of_number(8))
print(factorial_of_number(18))
print(factorial_of_number(45))


# Solution Two
def another_factorial(num):
  if num < 0:
    return 'Please provide a positive integer'
  if num ==1 or num ==0:
    return 1
  elif num >1:
    return num * another_factorial(num-1)

print(another_factorial(8))
print(another_factorial(18))
print(another_factorial(45))

#Solution Three
def third_factorial(num):
  if num < 0:
    return 'Please provide a positive integer'
  result = 1
  for i in range(1, num+1):
    result *= i
  return result

print(third_factorial(8))
print(third_factorial(18))
print(third_factorial(45))